pygeoapi/provider/ogr.py

```diff
@@ -457,10 +457,16 @@ def _get_next_feature(self, layer, feature_id):
 
     def _ogr_feature_to_json(self, ogr_feature):
         geom = ogr_feature.GetGeometryRef()
+        
+        #convert curvepolygon to polygon as GeoJSON doesn't support curves
+        if geom.GetGeometryType() == osgeo_ogr.wkbCurvePolygon:
```

Comment:
Curious, do we need to consider other geometry types that may need to be forced into polygon geometries?

Reply:
Yes, complicated. Hence my question in #861: *I see that there is a whole bunch:
https://gdal.org/development/rfc/rfc49_curve_geometries.html* . I  am no expert on this but:

* is this PR solving only a single use-case (single non-linear geometry)?
* can this be generically solved via an OGR option setting?